src/instructlab/model/train.py

```diff
@@ -629,8 +635,7 @@ def get_files(directory: str, pattern: str) -> list[str]:
 
         # pull the trainrandom.randinting and torch args from the flags
         # the flags are populated from the config as a base.
-        params = ctx.params
-        train_args, torch_args = map_train_to_library(params)
```

Comment:
In case reviewers are wondering (like me) if this would require a new training library dependency bump - no, the `TrainingArgs` model is using `ignore` strategy for unknown attributes, so it will silently swallow it.

---

(Question) Should we inform users that this happened? (I guess we could inspect the model and see if the attribute is defined on it - and warn or fail otherwise?)

Reply:
That's an interesting thought. One convenience of having the TrainingArgs object work this way is that the contract between the training repo and this CLI repo is fairly flexible in terms of backwards-compatibility. I don't love that we ignore / accept stuff silently. There should at least be debugging logging that captures this behavior. Would you consider making an issue for this?

Reply:
`pydantic` [allows](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.extra) to control what happens with unknown fields, but it has no mode that would e.g. "log warnings". If we want to deal with this situation, we will have to inspect model schemas ourselves. That's the price of loosely typed interface...

Or we just bump version of the library and be done with it. Or is it not an option because we are in blockers-only mode?

Reply:
Reported [this](https://github.com/instructlab/instructlab/issues/2019).